hazelcast/src/main/java/com/hazelcast/internal/util/BasicSymmetricCipherBuilder.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util;
+
+import com.hazelcast.config.AbstractBasicSymmetricEncryptionConfig;
+import com.hazelcast.internal.memory.impl.EndiannessUtil;
+import com.hazelcast.nio.Bits;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.DESKeySpec;
+import javax.crypto.spec.DESedeKeySpec;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.GeneralSecurityException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.KeySpec;
+
+import static com.hazelcast.internal.memory.impl.EndiannessUtil.BYTE_ARRAY_ACCESS;
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+public class BasicSymmetricCipherBuilder {
```

Comment:
Can we move this builder and the test into EE?

Reply:
Yes, we could, and I was actually thinking about that, too. It would have to reside in the hot-restart module but if that is fine, I can do that.

Reply:
Moved to the hot-restart module.